os/pm/pm_idle.c

```diff
@@ -186,7 +186,9 @@ void pm_idle(void)
 #ifdef CONFIG_SMP
 	/* Check if any core is gated, resume it */
 	while (gated_cpu_count) {
-		up_set_gating_flag_status(gated_cpu_count, 0);
+		if (up_get_cpu_state(gated_cpu_count) == CPU_RUNNING) {
```

Comment:
Is there other possible state that a gated core can obtain? If yes, there should be else statement to handle that condition. If not, we don't need to add this if condition.

Reply:
we are only doing gating at `CPU_RUNNING` state only. Before resuming CPU from gating, we should check it states is intact i.e. `CPU_RUNNING` state. Yes, we should handle else condition also.

Reply:
Done